src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -517,6 +557,36 @@ func (h *PromWriteHandler) parseRequest(
 		}
 	}
 
+	if h.remoteWriteOpts.rejectOldSamples {
+		now := h.nowFn()
+		thresholdTime := now.Add(-h.remoteWriteOpts.rejectDuration)
+		// The two for loops do the trick of not allocating new memory and do in place drops
+		i := 0
+		for _, ts := range req.Timeseries {
+			j := 0
+			var t int64
+			for _, s := range ts.Samples {
+				t = s.Timestamp
+				if time.UnixMilli(s.Timestamp).After(thresholdTime) {
+					// preserve samples that are new by reusing the same memory
+					ts.Samples[j] = s
+					j++
+				}
+			}
+			h.metrics.writeRejectTooOld.Inc(int64(len(ts.Samples) - j))
+			if j > 0 {
+				ts.Samples = ts.Samples[:j]
+				req.Timeseries[i] = ts
+				i++
+			} else if rand.Float32() < h.remoteWriteOpts.errorSamplingRate {
```

Comment:
this is sampling error logs

Reply:
this more like a warning log, minor: beside the sampling, can we also have a summary log for this request, how many we dropped in total

Reply:
I have a counter for this called `writeRejectTooOld`, log is time series level for vis, but metric shall be sample level since we drop based on timestamp of the samples.